jbpm-bpmn2/src/main/java/org/jbpm/bpmn2/xml/TaskHandler.java

```diff
@@ -270,7 +270,7 @@ public Object end(final String uri, final String localName,
 				String uniqueId = (String) node.getMetaData().get("UniqueId");
 				forEachNode.setMetaData("UniqueId", uniqueId);
 				node.setMetaData("UniqueId", uniqueId + ":" + uniqueIdGen++);
-				node.setMetaData("hidden", true);
+//				node.setMetaData("hidden", true);
```

Comment:
not really, it was incorrect behavior as it was skipping tasks within for each - multi instance on task (not subprocess) and by that process event listeners where not invoked.